src/core/HorizontalTabs/HorizontalTabs.tsx

```diff
@@ -143,7 +150,55 @@ export const HorizontalTabs = (props: HorizontalTabsProps) => {
     if (onTabSelected) {
       onTabSelected(index);
     }
-    setCurrentIndex(index);
+    setCurrentActiveIndex(index);
+  };
+
+  const onKeyDown = (event: React.KeyboardEvent<HTMLUListElement>) => {
+    // alt + left/right is used by browser / assistive technologies
+    if (event.altKey) {
+      return;
+    }
+
+    const isTabDisabled = (index: number) => {
+      const tab = labels[index];
+      return React.isValidElement(tab) && tab.props.disabled;
+    };
+
+    const focusTabAt = (index: number) => {
+      setFocusedIndex(index);
+      focusActivationMode === 'auto' && onTabClick(index);
+    };
+
+    let newIndex = focusedIndex ?? currentActiveIndex;
+    switch (event.key) {
+      case 'ArrowRight': {
+        do {
+          newIndex = (newIndex + 1 + labels.length) % labels.length;
```

Comment:
when you go right from the latest ,it should go to first? not sure...

Reply:
That is the W3 recommendation